### PR TITLE
Add requirements, design and spec documents for file verification in dry-run mode

### PR DESCRIPTION
**Pull request by isseis**
Requirements
- Add a detailed requirements document describing warn-only file verification in dry-run.
- Define DryRunResult file_verification fields and FileVerificationSummary structure.
- Specify logging behavior, JSON/TEXT output formats, and verification failure levels.
- Cover non-functional requirements, backward-compatibility, test expectations, and risks.

Design and spec 
- Add FileVerificationSummary and ResultCollector design to collect dry-run verification results
- Extend Verification Manager to support dry-run warn-only verification and result recording
- Add DryRunResult.FileVerification integration and extend formatters (Text/JSON)
- Add unit, integration, and E2E tests plus performance benchmarks; fix trailing whitespace

### CODE REVIEW

File: docs/tasks/0059_dryrun_file_verification/01_requirements.md [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+
+```json
+{
+  "metadata": { ... },
+  "status": "success",
+  "file_verification": {
+    "total_files": 10,
+    "verified_files": 8,
+    "skipped_files": 0,
+    "failed_files": 2,
```
[the remark this answers is not in the document]
Comment (isseis) [reply docs/tasks/0059_dryrun_file_verification/01_requirements.md]: done

File: docs/tasks/0059_dryrun_file_verification/03_specification.md [hunk 2]
```diff
@@ -0,0 +1,1464 @@
...
+## 8. セキュリティ仕様
+
+### 8.1 センシティブ情報のマスキング
+
+**ファイルパスのマスキング:**
+```go
+// maskSensitivePath implementation (already shown in Formatter section)
+// Example:
+// Input:  /home/user/secrets/config.json
+// Output: /***/secrets/config.json (when ShowSensitive=false)
```
[the remark this answers is not in the document]
Comment (isseis) [reply docs/tasks/0059_dryrun_file_verification/03_specification.md]: It doesn't make much sense to mask path names. Removed maskSensitivePath().

File: docs/tasks/0059_dryrun_file_verification/02_architecture.md [hunk 3]
```diff
@@ -0,0 +1,907 @@
...
+    D --> D2[runner --dry-run<br/>Hash File Not Found]
+    D --> D3[runner --dry-run<br/>Hash Mismatch]
+    D --> D4[runner --dry-run<br/>All Success]
+
+    style B fill:#e3f2fd
+    style C fill:#fff9c4
+    style D fill:#c8e6c9
+```
+
+### 9.2 主要テストケース
```
[the remark this answers is not in the document]
Comment (isseis) [reply docs/tasks/0059_dryrun_file_verification/02_architecture.md]: done

File: docs/tasks/0059_dryrun_file_verification/02_architecture.md [hunk 4]
```diff
@@ -0,0 +1,907 @@
...
+    section Phase 3
+    DryRunResult Extension            :p3, after p2t, 1d
+    Formatter Extension               :p3f, after p3, 2d
+    Format Tests                      :p3t, after p3f, 1d
+    section Phase 4
+    E2E Tests                         :p4, after p3t, 2d
+    Documentation                     :p4d, after p4, 1d
+```
+
+### 10.2 フェーズ詳細
```
[the remark this answers is not in the document]
Comment (isseis) [reply docs/tasks/0059_dryrun_file_verification/02_architecture.md]: done
